src/main/resources/category/java/common.xml

```diff
@@ -823,6 +823,40 @@ ancestor::MethodDeclaration//PrimarySuffix/@Image="disableConnectionState")]
         </properties>
     </rule>
 
+    <rule name="HttpClientBuilderWithoutPoolSize" class="net.sourceforge.pmd.lang.rule.XPathRule" deprecated="false" dfa="false" language="java" message="HttpClient builder is used and setMaxConnTotal and/or setMaxConnPerRoute is not called. HttpClient defaults of 2 per route and 20 max total are probably not right." typeResolution="true"
+          externalInfoUrl="http://www.jpinpoint.com/doc/Java+Code+Performance#JavaCodePerformance-IBI03">
+        <description>Problem: default there are only 2 connections per route available. When two external calls are being made, a third thread requesting a connection must wait for free connection.&#13;
+            Solution: HttpClients should explicitly define the number of connections per route. If only one route is used make the max number of connections equal to conn per route. (jpinpoint-rules)</description>
+        <priority>2</priority>
+        <properties>
+            <property name="version" value="2.0"/>
+            <property name="xpath">
+                <value><![CDATA[
+(:locally created http client builder without setMaxConnPerRoute :)
+//MethodDeclaration//PrimaryExpression/PrimaryPrefix/Name[
+(@Image="HttpClientBuilder.create" or @Image="HttpAsyncClientBuilder.create" or @Image="HttpClients.custom") and not(
+ancestor::MethodDeclaration//PrimarySuffix/@Image="setMaxConnPerRoute")]
+,
+(:locally created http client builder without setMaxConnTotal :)
+//MethodDeclaration//PrimaryExpression/PrimaryPrefix/Name[
+(@Image="HttpClientBuilder.create" or @Image="HttpAsyncClientBuilder.create" or @Image="HttpClients.custom") and not(
+ancestor::MethodDeclaration//PrimarySuffix/@Image="setMaxConnTotal")]
+,
+(: method param http client builder without setMaxConnPerRoute :)
+//MethodDeclaration//FormalParameter//ClassOrInterfaceType[
+(@Image="HttpClientBuilder" or @Image="HttpAsyncClientBuilder" or @Image="HttpClients") and not(
+ancestor::MethodDeclaration//PrimarySuffix/@Image="setMaxConnPerRoute")]
+,
+(: method param http client builder without setMaxConnPerRoute :)
+//MethodDeclaration//FormalParameter//ClassOrInterfaceType[
+(@Image="HttpClientBuilder" or @Image="HttpAsyncClientBuilder" or @Image="HttpClients") and not(
+ancestor::MethodDeclaration//PrimarySuffix/@Image="setMaxConnTotal")]
+			]]>
+                </value>
+            </property>
+        </properties>
+    </rule>
+
```

Comment:
Please use pmd-java:typeIs() for an exact match of the type.

Reply:
done